modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -228,7 +228,7 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>): void => {
     }
 
     // If after the last element in block node edge case for #5091
-    if (container.parentNode === parentBlock && isAfterLastNodeInContainer && !start) {
+    if ((container.parentNode === parentBlock || container === parentBlock) && isAfterLastNodeInContainer && !start) {
```

Comment:
So this change will alter any block element it seems that the bug also happens on h1 etc.

Reply:
no it doesn't happen for `h1`.
that `container` is [container = rng.startContainer](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/newline/InsertBlock.ts#L309), but for `h1` that `rng.startContainer` is the `h1` for `pre` is the `body`.

this is because before the [normalization](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/newline/InsertBlock.ts#L304), `rng.startContainer` is the `body`.

That normalization when you have a `pre` you get a different behavior from `h1` etc. because of [this check](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/selection/NormalizeRange.ts#L203), the `nonEmptyElementsMap` have these tags:
```
"summary", "textarea", "svg", "pre", "code", "script", "object", "audio", "video", "iframe", "th", "td", "track", "wbr", "source", "embed", "param", "meta", "link", "isindex", "input", "img", "hr", "frame", "col", "br", "basefont", "base", "area"
```

Reply:
I get an error on Chrome Mac if I run this fiddle that inserts a H1 with a text node and hit enter inside that H1: https://fiddle.tiny.cloud/sbZUY0pIPm/0

Reply:
I don't get any error in Chrome Linux :/ , does this branch solve the problem with `h1` on your machine?